### ISSUE TITLE
Disabling nc-httpsonly does not have any effect, still can't connect via http

### ISSUE THREAD

**Issue by briankendall** [nextcloud/nextcloudpi#1322]
Quite simply, I want to disable https-only because I will be using a forward proxy to handle all https to my Nextcloud instance (and other webservers). However after disabling nc-httpsonly it still redirects me every time I try to connect using just http.

**System information**

<--! Paste this in GitHub report -->

<details>
<summary>NextCloudPi diagnostics</summary>

```

NextCloudPi version  v1.38.5
NextCloudPi image    NextCloudPi_docker_07-20-19
distribution         Debian GNU/Linux 10 \n \l
automount            no
USB devices          none
datadir              /data/nextcloud/data
data in SD           yes
data filesystem      ext2/ext3
data disk usage      55G/472G
rootfs usage         55G/472G
swapfile             /dev/sda2
dbdir                /data/database
Nextcloud check      ok
Nextcloud version    19.0.13.1
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
Postfix service      up
internet check       ok
port check 80        open
port check 443       open
IP                   ***REMOVED SENSITIVE VALUE***
gateway              ***REMOVED SENSITIVE VALUE***
interface            eth0
certificates         ***REMOVED SENSITIVE VALUE***
NAT loopback         no
uptime               2:47

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "localhost",
            "11": "73.219.140.5",
            "1": "172.17.0.2",
            "5": "nextcloudpi.local",
            "7": "nextcloudpi",
            "8": "nextcloudpi.lan",
            "6": "nextcloud-bri.1337.cx",
            "3": "nextcloud.briankendall.net",
            "2": "nextcloud.briankendall.net",
            "4": "nextcloud.briankendall.net",
            "12": "nextcloud.briankendall.net"
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "19.0.13.1",
        "overwrite.cli.url": "https:\/\/nextcloud.briankendall.net\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "tempdirectory": "\/var\/www\/nextcloud\/data\/tmp",
        "mail_smtpmode": "sendmail",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "jpeg_quality": "60",
        "overwriteprotocol": "https",
        "maintenance": false,
        "data-fingerprint": "f392db3bba68967e5d4434f6253c682b",
        "theme": "",
        "loglevel": 0,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Mon Aug 23 00:24:38.631741 2021] [authz_host:error] [pid 99:tid 140547244164864] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/loading-small.gif failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.631941 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /css/ncp.css failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.632494 2021] [authz_host:error] [pid 99:tid 140547227363072] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/ncp-logo.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.654219 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/toggle.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.654386 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/defaults.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.654534 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/reboot.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.654661 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/search.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.656037 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/poweroff.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.656325 2021] [authz_host:error] [pid 99:tid 140547202160384] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/nc-button.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.656933 2021] [authz_host:error] [pid 99:tid 140547235763968] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/dashboard.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.657560 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/defaults-white.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.659018 2021] [authz_host:error] [pid 99:tid 140547202160384] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/info-white.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.659128 2021] [authz_host:error] [pid 99:tid 140547235763968] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/settings-white.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.659209 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/wizard.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.659656 2021] [authz_host:error] [pid 99:tid 140547227363072] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/files.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.827982 2021] [authz_host:error] [pid 99:tid 140547227363072] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:38.828409 2021] [authz_host:error] [pid 99:tid 140547252565760] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /ncp-output.php failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:40.438558 2021] [authz_host:error] [pid 99:tid 140547235763968] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /img/info.svg failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:46.876692 2021] [authz_host:error] [pid 99:tid 140547227363072] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Mon Aug 23 00:24:46.892871 2021] [authz_host:error] [pid 99:tid 140547227363072] [client 10.0.2.2:54437] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name

```
</details>

<details>
<summary>Database logs</summary>

```

2021-08-23  0:06:07 0 [Note] InnoDB: 10.3.29 started; log sequence number 41752217771; transaction id 35582813
2021-08-23  0:06:07 0 [Note] InnoDB: Loading buffer pool(s) from /data/database/ib_buffer_pool
2021-08-23  0:06:07 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-08-23  0:06:07 0 [Note] Recovering after a crash using tc.log
2021-08-23  0:06:07 0 [Note] Starting crash recovery...
2021-08-23  0:06:07 0 [Note] Crash recovery finished.
2021-08-23  0:06:07 0 [Note] Server socket created on IP: '127.0.0.1'.
2021-08-23  0:06:08 0 [Note] Reading of all Master_info entries succeeded
2021-08-23  0:06:08 0 [Note] Added new Master_info '' to hash table
2021-08-23  0:06:08 0 [Note] mysqld: ready for connections.
Version: '10.3.29-MariaDB-0+deb10u1'  socket: '/run/mysqld/mysqld.sock'  port: 3306  Debian 10
2021-08-23  0:06:16 0 [Note] InnoDB: Buffer pool(s) load completed at 210823  0:06:16
2021-08-23  0:06:17 20 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 21 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 22 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 23 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 24 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 25 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 26 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)
2021-08-23  0:06:17 27 [Warning] Access denied for user 'ncadmin'@'localhost' (using password: YES)

```
</details>

<details>
<summary>Nextcloud logs</summary>

```

{"reqId":"YSLuq9Q9fIdHtkkXiDyqDwAAAEY","level":1,"time":"2021-08-23T00:41:15+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLuqz4QhJcy5ncb-xz6gQAAABE","level":1,"time":"2021-08-23T00:41:15+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLuvNQ9fIdHtkkXiDyqEQAAAFU","level":1,"time":"2021-08-23T00:41:32+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLuvD4QhJcy5ncb-xz6iAAAERE","level":1,"time":"2021-08-23T00:41:32+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLuzNQ9fIdHtkkXiDyqGAAAAEI","level":1,"time":"2021-08-23T00:41:48+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLuzNQ9fIdHtkkXiDyqGQAASwU","level":1,"time":"2021-08-23T00:41:48+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu3dQ9fIdHtkkXiDyqGwAAAFE","level":1,"time":"2021-08-23T00:42:05+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu3T4QhJcy5ncb-xz6kQAABRE","level":1,"time":"2021-08-23T00:42:05+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu7NQ9fIdHtkkXiDyqIwAAAFA","level":1,"time":"2021-08-23T00:42:20+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu7D4QhJcy5ncb-xz6lgAAARE","level":1,"time":"2021-08-23T00:42:20+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu@9Q9fIdHtkkXiDyqJAAAAFI","level":1,"time":"2021-08-23T00:42:35+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLu@9Q9fIdHtkkXiDyqJQAAVwU","level":1,"time":"2021-08-23T00:42:35+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvCtQ9fIdHtkkXiDyqLQAAAFQ","level":1,"time":"2021-08-23T00:42:50+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvCj4QhJcy5ncb-xz6ngAADhE","level":1,"time":"2021-08-23T00:42:50+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvGdQ9fIdHtkkXiDyqLgAAAEs","level":1,"time":"2021-08-23T00:43:05+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvGT4QhJcy5ncb-xz6ogAABBE","level":1,"time":"2021-08-23T00:43:05+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvKdQ9fIdHtkkXiDyqOQAAAEM","level":1,"time":"2021-08-23T00:43:21+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvKT4QhJcy5ncb-xz6qQAAFxE","level":1,"time":"2021-08-23T00:43:21+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvONQ9fIdHtkkXiDyqOgAAAFY","level":1,"time":"2021-08-23T00:43:36+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:80\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}
{"reqId":"YSLvONQ9fIdHtkkXiDyqOwAARgU","level":1,"time":"2021-08-23T00:43:36+00:00","remoteAddr":"10.0.2.2","user":"--","app":"core","method":"GET","url":"/","message":"Trusted domain error. \"10.0.2.2\" tried to access using \"mediaserver.local:443\" as host.","userAgent":"AutoMounter/91 CFNetwork/978.0.7 Darwin/18.7.0 (x86_64)","version":"19.0.13.1"}

```

**Comment by nachoparker** [comment]
Hi,

Thanks for reporting. I just tested this and it works as intended.

You can check `/etc/apache2/sites-available/000-default.conf`. It should read `RewriteEngine Off` if httpsonly is disabled

**Comment by briankendall** [comment]
@nachoparker I looked into it a bit more, and I think I found the source of the problem, at least on my installation:

I needed to edit: `/data/nextcloud/config/config.php` and change line 52 from: `'overwriteprotocol' => 'https',` to `'overwriteprotocol' => 'http',`.

So long as the value was `https` it would always redirect from http to https, regardless of how `nc-httpsonly` was set. Once I made the change then `nc-httpsonly` works as expected.
